loom/engine/bindings/loom/lmSQLite.cpp

```diff
@@ -0,0 +1,467 @@
+        if(name == NULL)
+        {
+            lmLogError(gSQLiteGroup, "Invalid name for getParameterIndex in database: %s", parentDB->getDBName());
+        }
+        return index;
+    }
+
+    int bindInt(int index, int value)
+    {
+        int result = sqlite3_bind_int(statementHandle, index, value);
+        if(result != SQLITE_OK)
+        {
+            lmLogError(gSQLiteGroup, "Error calling bindInt for database: %s with Result Code: %i", parentDB->getDBName(), result);
+        }
+        return result;        
+    }
+
+    int bindDouble(int index, double value)
+    {
+        int result = sqlite3_bind_double(statementHandle, index, value);
+        if(result != SQLITE_OK)
+        {
+            lmLogError(gSQLiteGroup, "Error calling bindDouble for database: %s with Result Code: %i", parentDB->getDBName(), result);
+        }
+        return result;        
+    }
+
+    int bindString(int index, const char *value)
+    {
+        int result = sqlite3_bind_text(statementHandle, index, value, -1, SQLITE_TRANSIENT); 
+        if(result != SQLITE_OK)
+        {
+            lmLogError(gSQLiteGroup, "Error calling bindString for database: %s with Result Code: %i", parentDB->getDBName(), result);
+        }
+        return result;        
+    }
+
+    int bindBytes(int index, utByteArray *value)
+    {
+        void *bytes;
+        int size;
+
+        if(!value || !value->getSize())
+        {
+            bytes = NULL;
+            size = 0;
+        }
+        else
+        {
+            bytes = value->getDataPtr();      
+            size = (int)value->getSize();
+        }
+
+        //bind the blob to the statement
+        int result = sqlite3_bind_blob(statementHandle, index, (const void *)bytes, size, SQLITE_TRANSIENT); 
+        if(result != SQLITE_OK)
+        {
+            lmLogError(gSQLiteGroup, "Error calling bindBytes for database: %s with Result Code: %i", parentDB->getDBName(), result);
+        }
+        return result;
+    }
+
+    int step()
+    {
+        int result = sqlite3_step(statementHandle); 
+        if(result == SQLITE_ERROR)
+        {
+            lmLogError(gSQLiteGroup, "Error calling step for database: %s", parentDB->getDBName());
+        }
+        return result;
+    }
+
+    void stepAsync(lua_State *L)
+    {
+//BEN QUESTION: I need L in order to lualoom_pushnative<> the Statement into the Delegate 
+//              in the thread onComplete... but obviously it doen't always remain valid from 
+//              this call until the time for it to be used at the end of the stepAsync() thread... 
+//              what other options are there for me to pass the Statement through to the Delegate?
+//
+//              NOTE: This wont be an issue if we can nuke the Statement from needing to be passed
+//                    into the onStatementComplete delegate... which i'm still not sure why it is
+//                    done there but not for onStatementProgress...
+//
+//HACK: NOT STABLE!!!!
```

Comment:
The Delegate should automatically associate with L when someone adds a callback and it has a method to get L back in that case... Would that solve the issue here?